examples/torch/object_detection/layers/modules/ssd_head.py

```diff
@@ -90,6 +90,9 @@ def forward(self, features, image_tensor):
 
         with no_nncf_trace():
             priors = self.prior_box(features, image_tensor).to(loc.device)
+        # Through usage of Knowledge Distillation Algo priors were incorrectly differentiated what was causing on error.
+        # Non differential model outputs should be marked by requires_grad=False.
```

Comment:
Sorry for being picky, but I don't like this message.
Maybe in the next PR, but please try to explain the assumptions of KD about model outputs in more details. 
After reading this message still not clear the root cause of issue. 
Also it makes sense explain why you are doing detach instead of requires_grad=True

please create task for it, if you are not going to close it in this PR

@vanyalzr what do you think? does it make sense from your point of view?

Reply:
Maybe we could explain the assumptions of KD about model outputs in more general way than in `ssd_head.py`. So it would be a part of more general docs writing task. For now just leave like this or maybe even delete KD reference as explanation of `priors = priors.detach()`